### PR TITLE
Drone hatches

### PR DESCRIPTION
**Pull request by ike709**
:cl: Aurora, ported by ike709
add: Added drone hatches to most airlocks. Now maintenance drones shouldn't let people into places anymore.
/:cl:

I forgot to record audio, but here's some videos anyways from the FTL13 PR.
https://www.dropbox.com/s/d4li9ksrqgnpwdg/2017-10-15-0148-49.mp4?dl=0
https://www.dropbox.com/s/xeg5l3vuq4avu8l/2017-10-15-0149-37.mp4?dl=0

Tested and working.

Now if a drone is opening an airlock with a person nearby, you know it's to let them in/out!

### CODE REVIEW

**Comment by FlattestGuitar** [comment]
I think the sprites are garbage as well, but the door sprites suck anyway.

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -142,9 +151,34 @@
...
+	update_icon()
 
+/obj/machinery/door/airlock/proc/setup_hatch()
+	hatch_image = image('icons/obj/doors/hatches.dmi', src, "hatch_closed", layer+0.01)
+	hatch_image.color = hatch_colour
+	hatch_image.pixel_x = hatch_offset_x
+	hatch_image.pixel_y = hatch_offset_y
 	update_icon()
 
+/obj/machinery/door/airlock/proc/open_hatch(var/atom/mover = null)
```
Comment (FlattestGuitar) [inline code/game/machinery/doors/airlock.dm]: this whole thing could just be a `flick_overlay()`

**Comment by Ralta** [comment]
Idea is great, sprites aren't as has been said already.

**Comment by SamHPurp** [comment]
Idea is great, but I really don't like that the hatches entirely disappear when the door is in the opening animation.

**Comment by ike709** [comment]
Again... For all sprite complains, talk to Andrew.

**Comment by ike709** [comment]
@FlattestGuitar reeeeee

**Comment by FlattestGuitar** [comment]
Fuck it. The feature is helpful and there are much worse graphical atrocities all over the codebase.

I don't believe this is an issue at all myself, though it'd be nice to address this once we get to redoing door sprites.

**Pull request merged**